commit/merkleroot/types.go

```diff
@@ -173,3 +176,23 @@ const (
 	BuildingReport
 	WaitingForReportTransmission
 )
+
+type RootSet struct {
+	roots []cciptypes.MerkleRootChain
+}
+
+// Add a MerkleRoot to the set (if it's not already in the set)
+func (s *RootSet) Add(root cciptypes.MerkleRootChain) {
+	if !s.Contains(root) {
```

Comment:
`slices.Contains()` will do exactly this, right? Also if you want to have unique values here then mapset or map would be better probably and don't care about order of insertion then mapset or map would be better choice. Insertion when calling contains on a slice gonna have O(n^2) complexity. Probably not a problem for small slices, but it is easy to forget about that when the slice grows over time.